go-controller/pkg/ovn/multihoming_test.go

```diff
@@ -116,6 +116,7 @@ func (em *secondaryNetworkExpectationMachine) expectedLogicalSwitchesAndPorts(is
 		nodeslsps := make(map[string][]string)
 		acls := make(map[string][]string)
 		var switchName string
+		switchNodeMap := make(map[string]*nbdb.LogicalSwitch)
```

Comment:
yeah we shouldn't rely on that switch anyway, it's not logically connected (not directly anyways) 
So I added a new map to make things more clear.

Reply:
`switchNodeMap` map is used to insert the logicalMap **after** all passing all pods in the loop (filling the ports of every pod). only after passing over all the pods will the switchNodeMap be ready to be inserted to the `data` struct.
In an unrelated path, `alreadyAddedManagementElements` map holds entities that are inserted once per node AFAIK (like management Port, L3GWEntities, EgressEntities, etc..). The map is used in order to insert them **only once** . 
Maybe it'd be better to move that logic to a loop running over nodes. Would you prefer that?

Reply:
> `switchNodeMap` map is used to insert the logicalMap **after** all passing all pods in the loop (filling the ports of every pod). only after passing over all the pods will the switchNodeMap be ready to be inserted to the `data` struct.

OK, and you need this because the logical switch has a list of ports, right ? And now you'll have ports for all pods, instead of for a single pod. 

Right ? 

> 
> In a somewhat unrelated path, `alreadyAddedManagementElements` map holds entities that are inserted once per node AFAIK (like management Port, L3GWEntities, L2/L3EgressEntities, etc..). The map is used in order to insert them **only once** . Hope I got this clearer.
> 
> Maybe it'd be better to move that logic to a loop running over nodes. Would you prefer that?

I guess it's not needed.

Reply:
> > `switchNodeMap` map is used to insert the logicalMap **after** all passing all pods in the loop (filling the ports of every pod). only after passing over all the pods will the switchNodeMap be ready to be inserted to the `data` struct.
> 
> OK, and you need this because the logical switch has a list of ports, right ? And now you'll have ports for all pods, instead of for a single pod.

true

> 
> > In a somewhat unrelated path, `alreadyAddedManagementElements` map holds entities that are inserted once per node AFAIK (like management Port, L3GWEntities, L2/L3EgressEntities, etc..). The map is used in order to insert them **only once** . Hope I got this clearer.
> > Maybe it'd be better to move that logic to a loop running over nodes. Would you prefer that?
> 
> I guess it's not needed.

It's possible to keep a map[`<node-name>`][`struct-with-all-the-needed-info`] so I could construct it in another loop.. like I said this function is too complex for it's own good. de-clattering it needs thought cause this function is also used by the layer3 unit tests.. if you want this done on this PR rather than on a separate follow up PR then just say so..

Reply:
forget that. Too complex.

We should make these functions less convoluted eventually, because they're growing out of control.